pkg/infra/block_collector.go

```diff
@@ -11,39 +13,51 @@ import (
 // on a certain number of peers within network.
 type BlockCollector struct {
 	sync.Mutex
-	threshold int
-	total     int
-	registry  map[uint64]int
+	thresholdPeers int
+	totalPeers     int
+	totalTx        int
+	committedTx    int
+	registry       map[uint64]int
 }
 
 // NewBlockCollector creates a BlockCollector
-func NewBlockCollector(threshold int, total int) (*BlockCollector, error) {
+func NewBlockCollector(thresholdPeers int, totalPeers int, totalTx int) (*BlockCollector, error) {
 	registry := make(map[uint64]int)
-	if threshold > total {
-		return nil, errors.Errorf("threshold [%d] must be less than or equal to total [%d]", threshold, total)
+	if thresholdPeers > totalPeers {
+		return nil, errors.Errorf("threshold [%d] must be less than or equal to total [%d]", thresholdPeers, totalPeers)
 	}
-	return &BlockCollector{
-		threshold: threshold,
-		total:     total,
-		registry:  registry,
-	}, nil
+	bc := &BlockCollector{
+		thresholdPeers: thresholdPeers,
+		totalPeers:     totalPeers,
+		totalTx:        totalTx,
+		committedTx:    0,
+		registry:       registry,
+	}
+	return bc, nil
 }
 
 // Commit commits a block to collector. It returns true iff the number of peers on which
 // this block has been committed has satisfied threshold.
-func (bc *BlockCollector) Commit(block uint64) (committed bool) {
+func (bc *BlockCollector) Commit(block uint64, txs int, now time.Time) (committed bool) {
```

Comment:
it's not `now`, but `start`. also, why don't we pass this in at construction time?

furthermore, why couldn't we simply pass in `*Block` here?